adal/util.py

```diff
@@ -55,7 +55,11 @@ def add_default_request_headers(self, options):
     headers[AdalIdParameters.SKU] = AdalIdParameters.PYTHON_SKU
     headers[AdalIdParameters.VERSION] = adal.__version__
     headers[AdalIdParameters.OS] = sys.platform
-    headers[AdalIdParameters.CPU] = 'x64' if platform.architecture()[0] == '64bit' else 'x86'
+
+    try:
+        headers[AdalIdParameters.CPU] = 'x64' if platform.architecture()[0] == '64bit' else 'x86'
+    except:
+        headers[AdalIdParameters.CPU] = 'unknown'

```

Comment:
I can not speak for service side. Just thought we don't have to invent new value if we have other solutions.

In this particular case, that other solution is actually documented in [official document](https://docs.python.org/3/library/platform.html#cross-platform). And it IS the correct value because, as a scripting language running inside its interpretor, you can only know what architecture (64-bit or 32-bit) your interpretor supports. You don't know what the OS actually supports (and that doesn't mean anything to your current scripting environment anyway).

For example, I'm now using Windows 10, 64 bit, but install a 32-bit Python 3.5. Both the `platform.architecture()[0]` and `'x64' if sys.maxsize > 2**32 else 'x86'` give me a "x86" value, which is considered correct.